File: keycloak-client/src/main/java/cc/coopersoft/accesstoken/keycloak/model/KeycloakErrorResult.java
package cc.coopersoft.accesstoken.keycloak.model;

import cc.coopersoft.accesstoken.api.callback.ErrorResult;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public
class KeycloakErrorResult implements ErrorResult {

  public KeycloakErrorResult(String error) {
    this.error = error;
  }

  private String error;
  private String errorDescription;
}

File: keycloak-client/src/main/java/cc/coopersoft/accesstoken/keycloak/model/KeycloakErrorResult.kt
package cc.coopersoft.accesstoken.keycloak.model

import cc.coopersoft.accesstoken.api.callback.ErrorResult

class KeycloakErrorResult(override val error: String?, override val errorDescription: String?) :
    ErrorResult {
    constructor():this(null, null)
    constructor(error: String?) : this(error, null)
}